ignite/metrics/metric.py

```diff
@@ -108,14 +118,37 @@ def _sync_all_reduce(self, tensor: Union[torch.Tensor, numbers.Number]) -> Union
         else:
             raise TypeError("Unhandled input type {}".format(type(tensor)))
 
-        # synchronize and reduce
-        dist.barrier()
-        dist.all_reduce(tensor)
+        reduction(tensor)
 
         if tensor_to_number:
             return tensor.item()
         return tensor
 
+    @staticmethod
+    def _tpu_reduce(tensor: Union[torch.Tensor, numbers.Number]) -> None:
+        xm.all_reduce("sum", [tensor,])
+
+    @staticmethod
+    def _gpu_reduce(tensor: Union[torch.Tensor, numbers.Number]) -> None:
+        dist.barrier()
+        dist.all_reduce(tensor)
+
+    def _tpu_sync_all_reduce(self, tensor: Union[torch.Tensor, numbers.Number]) -> Union[torch.Tensor, numbers.Number]:
+        return self._do_reduction(tensor, self._tpu_reduce, torch.float)
+
+    def _gpu_sync_all_reduce(self, tensor: Union[torch.Tensor, numbers.Number]) -> Union[torch.Tensor, numbers.Number]:
+        return self._do_reduction(tensor, self._gpu_reduce)
+
+    def _sync_all_reduce(self, tensor: Union[torch.Tensor, numbers.Number]) -> Union[torch.Tensor, numbers.Number]:
+        if not (dist.is_available() and dist.is_initialized()) and not xrt_world_size > 0:
+            # Nothing to reduce
+            return tensor
+
+        if on_xla_device:
+            return self._tpu_sync_all_reduce(tensor)
```

Comment:
Can we directly call `self._do_reduction(tensor, self._gpu_reduce)` intead ?
The purpose of `_gpu_sync_all_reduce` and `_tpu_sync_all_reduce` is very limited for me...